### PR TITLE
Create the charmrepo package

### PR DESCRIPTION
**Pull request by frankban**
Move repo.go in the charmrepo package.
This is done to avoid cyclic imports later, when charmrepo will import from charmstore/csclient.

I did not stash the commits so that the changes are easier to follow.
Some of them are mechanical, some not.

Please check that it compiles and the tests pass.

### CODE REVIEW

**Comment by mhilton** [comment]
:+1: LGTM No QA.

File: charmrepo/repo.go [hunk 1]
```diff
@@ -51,15 +57,15 @@ type CharmRevision struct {
 
 // Repository represents a collection of charms.
 type Repository interface {
```
Comment (rogpeppe) [inline charmrepo/repo.go]: perhaps s/Repository/Interface/ ?

**Comment by rogpeppe** [comment]
LGTM with one thought above.

File: charmrepo/repo.go (hunk 1 above)
Comment (frankban) [inline charmrepo/repo.go]: Done.

**Pull request merged**